src/hydrusvideodeduplicator/dedup.py

```diff
@@ -145,52 +165,39 @@ def _add_perceptual_hashes_to_db(self, overwrite: bool, video_hashes: Sequence[s
         with SqliteDict(str(DEDUP_DATABASE_FILE), tablename="videos", flag="c") as hashdb:
             dblen = len(hashdb)
             dbsize = os.path.getsize(DEDUP_DATABASE_FILE)
+            videoslen = len(video_hashes)
 
             if dblen > 0:
                 self.hydlog.info(f"Database found of length {dblen}, size {dbsize} bytes")
             else:
                 self.hydlog.info(f"Database not found. Creating one at {DEDUP_DATABASE_FILE}")
 
             try:
-                with tqdm(total=len(video_hashes), dynamic_ncols=True, unit="video", colour="BLUE") as pbar:
-                    count_since_last_commit = 0
-                    COMMIT_INTERVAL = 16
-
-                    for video_hash in video_hashes:
-                        pbar.update(1)
-                        # Only calculate new hash if it's missing or if overwrite is true
-                        if not overwrite and video_hash in hashdb and "perceptual_hash" in hashdb[video_hash]:
-                            continue
-
-                        # Get video file from Hydrus
-                        try:
-                            video_response = self.client.get_file(hash_=video_hash)
-                        except hydrus_api.HydrusAPIException:
-                            rprint("[red] Failed to get video from Hydrus.")
-                            self.hydlog.error("Error getting video from Hydrus.")
-                            continue
-
-                        # Calculate perceptual_hash
-                        try:
-                            perceptual_hash = self._calculate_perceptual_hash(video_response.content)
-                        except Exception as exc:
-                            rprint("[red] Failed to calculate a perceptual hash.")
-                            self.hydlog.exception(exc)
-                            self.hydlog.error(f"Errored file hash: {video_hash}")
-                        else:
-                            # Write perceptual hash to DB
+                if overwrite or dblen == 0:
+                    hashes_to_process = video_hashes
+                else:
+                    # We could put a progress bar on this process, but it's only a few seconds on a 10k db right now
+                    hashes_to_process = [video_hash for video_hash in video_hashes if
```

Comment:
can we turn the video_hashes into a set when they're gotten from the api and then just remove them instead of creating another list using list comprehension? might be faster.

Reply:
@ianwal It would be more memory-efficient, but it would actually be slower - [because iterating over a list in Python is much faster than iterating over a set](https://stackoverflow.com/a/62563234).

Reply:
That link says that iterating over a list with 400 numbers would take **5 seconds**...

Try it if you can. I think it would work.

Reply:
Went with the approach we discussed! Good idea 👍🏼